### PR TITLE
Add Ref and Commit outputs

### LINKED ISSUE #281: Make checked out ref available to other steps
Pull requests events by default check out the merge commit ref. Alternatively, the pull request head can be checkout using:

```yaml
- uses: actions/checkout@v2
  with:
    ref: ${{ github.event.pull_request.head.sha }}
```

However, in successive actions it is not possible to detect whether the action uses another ref. `github.context.sha` still references the merge commit. See for example the discussion on: https://github.com/jwgmeligmeyling/pmd-github-action/issues/2

It would be nice if `actions/checkout@v2` somehow made the checkout ref available as output or exported variable.

Or am I overlooking an existing approach to obtain the checked out commit SHA?

### PR DESCRIPTION
**Pull request by lucacome**
Adds outputs for `ref` and `commit`.

Closes #281

### CODE REVIEW

**Review by Spiritaine — APPROVED** [review]

**Review by Spiritaine — APPROVED** [review]

**Comment by lucacome** [comment]
anybody around to approve the workflow run? and maybe review it? 😅

**Review by Spiritaine — APPROVED** [review]

**Comment by greenhat616** [comment]
Any updates here? 🚀

**Comment by Sushisource** [comment]
Yeah, a bit wild this has just been sitting around for over a year. There's really no one at GH/MS that can approve and merge this?

**Comment by rantoniuk** [comment]
@joshmgross sorry for pulling you in directly here, but I see you were preparing the last release. Would you mind looking at who could at least review this?
This looks like a basic functionality that should be there for checking out custom refs.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -295,3 +295,32 @@ jobs:
...
+        with:
+          ref: test-data/v2/basic
+
+      # Verify output
+      - name: Verify output
+        run: |
+          echo "Commit: ${{ steps.checkout.outputs.commit }}"
+          echo "Ref: ${{ steps.checkout.outputs.ref }}"
+
+          if [ "${{ steps.checkout.outputs.ref }}" != "test-data/v2/basic" ]; then
```
Comment (joshmgross) [inline .github/workflows/test.yml:319]: Can we add a similar check validating that the commit output matches https://github.com/actions/checkout/commit/82f71901cf8c021332310dcc8cdba84c4193ff5d?

Comment (lucacome) [reply .github/workflows/test.yml:319]: @joshmgross done!

**Comment by OldCrowEW** [comment]
hello, will this be in the next release? 🙏

**Comment by alex-dyla** [comment]
This would be great to have in!

**Comment by rantoniuk** [comment]
@joshmgross ping again, since the review comments were immediately resolved.

**Review by joshmgross — APPROVED** [review]

**Pull request merged**